### PR TITLE
Remove redundant condition

### LINKED ISSUE #2569: Remove "redundant" test condition in pcl_targets.cmake.
@jspricke Actually, do you see any reason for this test you referenced:

https://github.com/PointCloudLibrary/pcl/blob/a602b19cbb392bc16a26498443c3ba5010bd45f3/cmake/pcl_targets.cmake#L113

Why isn't the first part of the condition sufficient? Looking at the Git history, this was added as a single commit and does not provide any motivation. Maybe we just drop it? Or better after the release?

_Originally posted by @taketwo in https://github.com/PointCloudLibrary/pcl/pull/2563#issuecomment-431290683_

### PR DESCRIPTION
**Pull request by shrijitsingh99**
Resolves #2569

### CODE REVIEW

File: cmake/pcl_targets.cmake [hunk 1]
```diff
@@ -114,7 +114,7 @@ macro(PCL_SUBSYS_DEPEND _var _name)
     if(SUBSYS_EXT_DEPS)
       foreach(_dep ${SUBSYS_EXT_DEPS})
         string(TOUPPER "${_dep}_found" EXT_DEP_FOUND)
-        if(NOT ${EXT_DEP_FOUND} AND (NOT (${EXT_DEP_FOUND} STREQUAL "TRUE")))
+        if(NOT ${EXT_DEP_FOUND})
```
Comment (taketwo) [inline cmake/pcl_targets.cmake:117]:
```suggestion
if(NOT EXT_DEP_FOUND)
```

No need to explicitly apply variable expansion inside `if` statements.

**Comment by kunaltyagi** [comment]
Pinging due to no activity for 15 days (I'm the stale-bot 🤣 )

**Comment by shrijitsingh99** [comment]
> Pinging due to no activity for 15 days (I'm the stale-bot rofl )

Sorry about the delay :sweat_smile:. I will get this moving ASAP.
I have a lot of PR's getting stale, will get some progress in all of them this week.

**Comment by kunaltyagi** [comment]
It doesn't work for downstream targets. Tutorial CI fails with almost all tutorials unable to find header files

**Comment by shrijitsingh99** [comment]
> It doesn't work for downstream targets. Tutorial CI fails with almost all tutorials unable to find header files

It was a typo issue from my side. Fixed now.

**Pull request merged**